artemis-wildfly-integration/src/main/java/org/jboss/activemq/artemis/wildfly/integration/WildFlySSLContextFactory.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.activemq.artemis.wildfly.integration;
+
+import static org.apache.activemq.artemis.spi.core.remoting.ssl.SSLContextFactory.log;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import javax.net.ssl.SSLContext;
+import org.apache.activemq.artemis.core.remoting.impl.netty.TransportConstants;
+import org.apache.activemq.artemis.core.remoting.impl.ssl.DefaultSSLContextFactory;
+import org.apache.activemq.artemis.spi.core.remoting.ssl.SSLContextConfig;
+import org.apache.activemq.artemis.utils.ConfigurationHelper;
+
+/**
+ *
+ * @author Emmanuel Hugonnet (c) 2019 Red Hat, Inc.
+ */
+public class WildFlySSLContextFactory extends DefaultSSLContextFactory {
+
+    private static final Map<String, SSLContext> SSL_CONTEXTS = Collections.synchronizedMap(new HashMap<String, SSLContext>());
+
+    public static void registerElytronSSLContext(String name, SSLContext context) {
```

Comment:
is there anything specific about Elytron in this class? afaict, this allows to pull the SSLContext from WildFly codebase but there is nothing that requires this context to be from Elytron. Is that correct?

Reply:
Correct